Assets/Scripts/Player/PlayerController.cs

```diff
@@ -1,93 +1,133 @@
+
+    private void Awake()
+    {
         rb2D = GetComponent<Rigidbody2D>();
         targetPosition = rb2D.position;
         currentHealth = maxHealth;
+        playerPos = this.GetComponent<Transform>();
+        playerColl = this.GetComponent<Collider2D>();
+        nextFire = Time.time;
     }
 
     public int GetMaxHealth() => maxHealth;
     public int GetCurrentHealth() => currentHealth;
 
-    private void Update() {
-        if (PauseController.Instance.GamePaused) {
-            return;
-        }
+    private void Update()
+    {
+        if (PauseController.Instance.GamePaused) { return; }
 
-        if (Input.touchCount > 0) {
+        if (Input.touchCount > 0)
+        {
             Touch touch = Input.GetTouch(0);
 
-            if (touch.phase == TouchPhase.Began && !EventSystem.current.IsPointerOverGameObject(touch.fingerId)) {
+            if (touch.phase == TouchPhase.Began && !EventSystem.current.IsPointerOverGameObject(touch.fingerId))
+            {
                 targetPosition = Camera.main.ScreenToWorldPoint(touch.position);
             }
         }
-        else if (Input.GetMouseButtonDown(0) && !EventSystem.current.IsPointerOverGameObject()) {
+        else if (Input.GetMouseButtonDown(0) && !EventSystem.current.IsPointerOverGameObject())
+        {
             targetPosition = Camera.main.ScreenToWorldPoint(Input.mousePosition);
         }
 
-        if (Input.GetKeyDown(KeyCode.H))
-        {
-            Heal(20);
-        }
+        if (rb2D.position == targetPosition && enemyHash.Count > 0) { RangedAttack(); }
+
+        if (Input.GetKeyDown(KeyCode.H)) { Heal(20); }
+
+        if (Input.GetKeyDown(KeyCode.Space)) { TakeDamage(20); }
 
-        if (Input.GetKeyDown(KeyCode.Space))
-        {
-            TakeDamage(20);
-        }
         if (GetCurrentHealth() <= 0)
         {
-            
             Scene currentScene = SceneManager.GetActiveScene();
             SceneManager.LoadScene(currentScene.name);
-          
         }
     }
 
-    private void FixedUpdate() {
+    private void FixedUpdate()
+    {
         Vector2 newPosition = Vector2.MoveTowards(rb2D.position, targetPosition, movementSpeed);
         rb2D.MovePosition(newPosition);
 
         // TODO: Maybe set targetPosition to currentPosition when the player is stopped. Then, the player
         // won't continually try to reach an unreachable targetPosition when blocked by a collider.
     }
 
-    public void TakeDamage(int damage)
+    private void RangedAttack()
     {
+        if (Time.time > nextFire)
+        {
+            Instantiate(projectile, transform.position, Quaternion.identity);
+            proj = projectile.GetComponent<ProjectileController>();
 
-        currentHealth -= damage;
-        healthBar.SetHealth(currentHealth);
+            if (proj != null)
+            {
+                proj.target = FindTargetEnemy();
+                proj.senderColl = playerColl;

```

Comment:
I think it would be good to add a `SetUpProjectile` method that sets these variables to `ProjectileController.cs`.
There are a few benefits to doing that:
- Less code is needed in scripts that spawn projectiles.
- These projectile variables can be private.
- If another projectile variable needs to be set on projectile spawn in the future, it's easier to ensure that scripts spawning projectiles are setting all of the needed projectile variables by changing the signature of that method.
Adding that method also reduces the amount of code in and means